libcodechecker/server/client_db_access_handler.py

```diff
@@ -1148,40 +1358,201 @@ def getDiffResultTypes(self,
                            diff_type,
                            report_filters):
 
-        session = self.__session
-        base_line_hashes, new_check_hashes = \
-            self.__get_hashes_for_diff(session,
-                                       base_run_id,
-                                       new_run_id)
+        try:
+            session = self.__Session()
+            base_line_hashes, new_check_hashes = \
+                self.__get_hashes_for_diff(session,
+                                           base_run_id,
+                                           new_run_id)
+
+            if diff_type == DiffType.NEW:
+                diff_hashes = list(
+                    new_check_hashes.difference(base_line_hashes))
+                if not diff_hashes:
+                    return diff_hashes
+                run_id = new_run_id
+
+            elif diff_type == DiffType.RESOLVED:
+                diff_hashes = list(
+                    base_line_hashes.difference(new_check_hashes))
+                if not diff_hashes:
+                    return diff_hashes
+                run_id = base_run_id
+
+            elif diff_type == DiffType.UNRESOLVED:
+                diff_hashes = list(
+                    base_line_hashes.intersection(new_check_hashes))
+                if not diff_hashes:
+                    return diff_hashes
+                run_id = new_run_id
+
+            else:
+                msg = 'Unsupported diff type: ' + str(diff_type)
+                LOG.error(msg)
+                raise shared.ttypes.RequestFailed(
+                    shared.ttypes.ErrorCode.DATABASE,
+                    msg)
+
+            return self.__queryDiffResultTypes(session,
+                                               diff_hashes,
+                                               run_id,
+                                               report_filters)
+        finally:
+            session.close()
+
+    @timeit
+    def necessaryFileContents(self, file_hashes):
+        try:
+            session = self.__Session()
 
-        if diff_type == DiffType.NEW:
-            diff_hashes = list(new_check_hashes.difference(base_line_hashes))
-            if not diff_hashes:
-                return diff_hashes
-            run_id = new_run_id
+            q = session.query(FileContent) \
+                .options(sqlalchemy.orm.load_only('content_hash')) \
+                .filter(FileContent.content_hash.in_(file_hashes))
 
-        elif diff_type == DiffType.RESOLVED:
-            diff_hashes = list(base_line_hashes.difference(new_check_hashes))
-            if not diff_hashes:
-                return diff_hashes
-            run_id = base_run_id
+            return list(set(file_hashes) -
```

Comment:
Maybe this set difference could be calculated by the database engine instead of python?